### ISSUE TITLE
Feature request: let Variable Explorer look inside function environments

### ISSUE THREAD

**Issue by Hong-Revo** [Microsoft/RTVS#3540]
Say I have a function that defines its own enclosing environment:
``` r
f <- local({
    x <- 42   # local state
    function(y) {
        x <<- x + y
        x
    }
})

f(3)
## 45

environment(f)$x
## 45
```
It would be nice to be able to look inside `f`'s environment and pull out the value of `x`. We can already do something like this when debugging, and VE can also look inside raw environments, so hopefully it shouldn't be too difficult.

**Comment by int19h** [comment]
Yep, just need to add `environment(f)` as the child of any `f` that is a function.

In general, we need to add Watch window like support to Variable Explorer (such that you can type arbitrary experssions and view them there, not just variables it scrapes from environments). Then you could do this, and a lot more besides.
